app/main.py

```diff
@@ -1,6 +1,41 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    # Store cache per function
+    func_cache: dict[tuple, any] = {}
+
+    @wraps(func)
+    def wrapper(*args) -> any:
+        # Use the arguments as the cache key
+        cache_key = args
```

Comment:
You should check both `args` and `kwargs` because the function might accept the same `args` but different `kwargs`. Additionally, I recommend checking the function name since there could be two different functions that accept the same arguments.